### DISCUSSION TITLE
Custom Citation Key Help

### DISCUSSION THREAD

**Discussion by MemeAddict314** [retorquere/zotero-better-bibtex#2868] [Q&A] [answered]
Hi! I'm trying to format my citekeys but having issues with ternaries. I'm trying to make the citekey for encyclopedias (specifically Wikipedia) appear different from everything else. My current citekey looks like this:
`type(encyclopediaArticle) ? (shorttitle.capitalize + EncyclopediaTitle.capitalize) : (authEtAl.capitalize + year)`
It works right for most things, but for some reason it adds journal abbreviations in front of the title on some items. Right now I have these citekeys in these kinds of formats outputted by my citekey: 
- `Encyclopedia Name + Shorttitle` (correct)
- `Author + Year` (wrong with encyclopedias, right for other sources)
- `Journal Abbreviation/Publication + Shorttitle` (very wrong)

I've tried a bunch of different options, and I can't find documentation for what I'm looking for. What I'd like to happen would be something like: 
```
If item is (from Wikipedia or an encyclopedia)
Then EncyclopediaTitle.capitalize + shorttitle.capitalize
Else authEtAl.capitalize + year
```
I don't want the journal name, journal abbreviation, or publication name in the citekey. I want the name of the encyclopedia then the short title only for encyclopedia entries. I'm mainly focused on Wikipedia articles being cited and saved, but I'd prefer if the citekey worked on all encyclopedias. Thank you for any help you can give!

**Comment by retorquere** [comment]
I'll need a debug log for that item, then I'll be able to tell precisely.

**Reply by MemeAddict314** [reply]
For a Wikipedia article that incorrectly displays authorName + year: SJSCANL6-refs-euc/6.7.185-7
For a Wikipedia article that correctly displays encyclopediaName + shorttitle: 733HPIAA-refs-euc/6.7.185-7
For a paper that correctly displays authorName + year: HIFREUAC-refs-euc/6.7.185-7
For a paper that has incorrectly displayed publication/journalAbbreviation + year (but seems to be fixed now despite me not changing anything?): GXW2IFKG-refs-euc/6.7.185-7
Thank you!

**Reply by retorquere** [reply, marked as answer]
I'm going to take this up in #2872. It's a bug in BBT that's not a quick fix. For the time being, you can use

```
type(encyclopediaArticle) + shorttitle.capitalize + EncyclopediaTitle.capitalize; authEtAl.capitalize + year
```

**Reply by MemeAddict314** [reply]
Thank you!
